### ISSUE TITLE
There is some way to save the settings used in the photo to apply them automatically later?

### ISSUE THREAD

**Issue by janczakb** [GottaYotta/PixelSDK#8]
There is some way to save the settings used in the photo to apply them automatically later?

Saves photos before and after editing. Can I save all effects in a variable?

So that I can later use them on the original photo and get the effect of the edited photo?

**Comment by joshbernfeld** [comment]
This is possible. We will put together an example.

**Comment by joshbernfeld** [comment]
You will have to update to the newest version of PixelSDK (5.5.0) before using this sample code.
```
$ pod update
```
<br>
This displays an EditController with our saved filters (if any). The filters are saved whenever the Next button is pressed in the EditController.
<br>
<br>

```swift
class ViewController: UIViewController, EditControllerDelegate {

    // Saved primary filter from the Filter tab
    var savedPrimaryFilter: SessionFilter?
    // Saved filters from the Adjust tab
    var savedFilters = [SessionFilter]()
    
    @IBAction func displayEditController() {
        let session = Session(image: UIImage(named: "test_image")!)
        
        // Apply our saved filters.
        // Copy them otherwise the session might modify our saved filters.
        session.image!.primaryFilter = self.savedPrimaryFilter?.copy()
        session.image!.filters = self.savedFilters.copy()
        
        // Initialize the EditController
        let editController = EditController(session: session)
        editController.delegate = self
        self.present(editController, animated: true, completion: nil)
    }

     // Called when the Next button in the EditController is pressed.
    func editController(_ editController: EditController, didFinishEditing session: Session) {
        print("Save Filters")
        
        if let image = session.image {
            // Save our filters.
            // Copy them otherwise the session might modify our saved filters.
            self.savedPrimaryFilter = image.primaryFilter?.copy()
            self.savedFilters = image.filters.copy()
        }
    }

}
```
<br>
If you want your saved filter variables to persist after restart, you can replace them with the following. Not very pretty unfortunately. If you do use this, you no longer need to call copy() above.
<br>
<br>

```swift
var savedPrimaryFilter: SessionFilter? {
    get {
        if let data = UserDefaults.standard.data(forKey: "savedPrimaryFilter") {
            return try? SessionFilter.fromData(data)
        }
        return nil
    }
    set {
        UserDefaults.standard.set(try? newValue?.toData(), forKey: "savedPrimaryFilter")
    }
}

var savedFilters: [SessionFilter] {
    get {
        let dataArray = (UserDefaults.standard.array(forKey: "savedFilters") as? [Data]) ?? [Data]()
        return dataArray.map { (try? SessionFilter.fromData($0)) ?? SessionFilter() }
    }
    set {
        let dataArray = (try? newValue.map { try $0.toData() }) ?? [Data]()
        UserDefaults.standard.set(dataArray, forKey: "savedFilters")
    }
}
```

**Comment by joshbernfeld** [comment]
This bug has been fixed in the latest release (6.0.0). Please run `pod update`.
